src/js/index.js

```diff
@@ -56,38 +59,90 @@ RequestUtil.json = function (options = {}) {
         </Provider>),
         domElement);
     } else {
-      setTimeout(function () {
-        let routes = RouterUtil.buildRoutes(appRoutes.getRoutes());
-        let router = Router.run(routes, function (Handler, state) {
-          Config.setOverrides(state.query);
-          ReactDOM.render(
-            (<Provider store={PluginSDK.Store}>
-              <Handler state={state} />
-            </Provider>),
-            domElement);
+      let loadedResources = {
+        mesos: false
+      };
+      let mesosEvents = [MESOS_SUMMARY_CHANGE, MESOS_SUMMARY_REQUEST_ERROR];
+
+      // Responsible for checking if application is ready to render
+      function delayOrLoadApplication() {
+        // Check if we've loaded all resources first
```

Comment:
I'd suggest we have a `alreadyLoaded` flag so we don't run into accidentally calling `renderApplicationToDOM` more than once. If we have more than 1 item in `loadedResources` then this could happen because we're using `setTimeouts`.